src/aes_tools/mod.rs

```diff
@@ -45,12 +56,19 @@ impl Cipher for AESCipher {
 }
 
 pub fn generate_keys() -> (AesKey, AesKey) {
-    let buf = generate_buffer(16); // 128, 192, 256 bits or 16, 24, 32 bytes
+    let buf = generate_buffer(32); // 128, 192, 256 bits or 16, 24, 32 bytes
     let e_aeskey = AesKey::new_encrypt(&buf).expect("failed to generate encrypt key");
     let d_aeskey = AesKey::new_decrypt(&buf).expect("failed to generate decrypt key");
     (e_aeskey, d_aeskey)
 }
 
+pub fn generate_keys_from_shared_secret(secret: SharedSecret) -> (AesKey, AesKey) {
+    let buf = secret.as_bytes();
+    let e_aeskey = AesKey::new_encrypt(buf).expect("failed to generate encrypt key");
+    let d_aeskey = AesKey::new_decrypt(buf).expect("failed to generate decrypt key");
```

Comment:
Does the buffer move here? Should it be borrowed?

Reply:
It does pass tests. as_bytes() returns a value that implements Copy so in this context what we've got there should be fine.

Reply:
Ah, the copy trait. Yeah I figured it was fine since it passes the test, it just perplexed me at first. But yeah, if it has the copy trait, that is the part I was missing.